src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -508,7 +456,7 @@ internal virtual bool DoRemove(T item)
                             // If parent is a 3-node, flip the orientation of the red link. 
                             // We can achieve this by a single rotation        
                             // This case is converted to one of other cased below.
-                            Debug.Assert(!parent.IsRed, "parent must be a black node!");
+                            Debug.Assert(!parent.IsRed);
```

Comment:
None of the messages seem to add any value; if I'm reading the code and I see an assert that `parent != null`, I don't also need to be told that `"parent must not be null!"`. For the rare times that one of these asserts does fail, a line number will be shown in the stack trace so we won't rely on the message to identify which assert failed.